src/python/pants/backend/python/dependency_inference/import_parser.py

```diff
@@ -67,14 +67,15 @@ def visit_Call(self, node):
           and node.func.id == "__import__"
           and len(node.args) == 1
       ):
-          if sys.version_info[0:2] < (3, 8) and isinstance(node.args[0], ast.Str):
-              arg_s = node.args[0].s
-              val = arg_s.decode("utf-8") if isinstance(arg_s, bytes) else arg_s
-              self.imports.add(arg_s)
-              return
-          elif isinstance(node.args[0], ast.Constant):
-              self.imports.add(str(node.args[0].value))
-              return
+            if sys.version_info[0:2] < (3, 8):
```

Comment:
This will be easier to follow (and easier to reason about the correctness of) if you set `name = None` up front and then re-set it inside conditionals.

Reply:
That's how I had it at first, but it resulted in 5 levels of nesting from the original function iirc, so went for this.

Either way, this code is only on 2.9 and 2.10 code has diverged a lot already, so nbd